packages/core/src/services/documents/getDocumentMetadata.ts

```diff
@@ -1,47 +1,37 @@
 import path from 'path'
 
-import { readMetadata, Document as RefDocument } from '@latitude-data/compiler'
+import {
+  ConversationMetadata,
+  readMetadata,
+  Document as RefDocument,
+} from '@latitude-data/compiler'
 
 import { Commit, DocumentVersion, Workspace } from '../../browser'
-import {
-  LatitudeError,
-  Result,
-  TypedResult,
-  UnprocessableEntityError,
-} from '../../lib'
+import { LatitudeError, NotFoundError, Result, TypedResult } from '../../lib'
 import { DocumentVersionsRepository } from '../../repositories'
 
 /**
  * This is an internal method. It should always receives
  * workspaceId from a trusted source. Like for example API gateway that validates
  * requested documents belongs to the right workspace.
  */
-export async function getResolvedContent({
-  workspaceId,
+export async function getDocumentMetadata({
+  workspace,
   document,
   commit,
 }: {
-  workspaceId: Workspace['id']
+  workspace: Workspace
   document: DocumentVersion
   commit: Commit
-}): Promise<TypedResult<string, LatitudeError>> {
-  const documentScope = new DocumentVersionsRepository(workspaceId)
+}): Promise<TypedResult<ConversationMetadata, LatitudeError>> {
+  const documentScope = new DocumentVersionsRepository(workspace.id)
   const docs = await documentScope
     .getDocumentsAtCommit(commit)
     .then((r) => r.unwrap())
-
   const docInCommit = docs.find((d) => d.documentUuid === document.documentUuid)
-
   if (!docInCommit) {
-    return Result.error(
-      new UnprocessableEntityError('Document not found in commit', {}),
-    )
+    return Result.error(new NotFoundError('Document not found in commit'))
   }
-
-  if (commit.mergedAt != null && document.resolvedContent != null) {
-    return Result.ok(document.resolvedContent!)
-  }
```

Comment:
I removed this early return which is a small optimization to avoid computing the metadata of a document, we always want to compute the metadata in order to know if the response is cached or not

Reply:
yeah I think it's worth it when we trade it off with the cache